### PR TITLE
QDB-9669 - Build with centos7.

### PR DESCRIPTION
**Pull request by vianneyPL**

### CODE REVIEW

File: scripts/teamcity/10.build.sh [hunk 1]
```diff
@@ -1,6 +1,16 @@
+#!/bin/bash
+
+# Using cento7 docker we need to enable gcc-8 and g++-8
```
Comment (mkurdej) [inline scripts/teamcity/10.build.sh:3]:
```suggestion
# Using centos7 docker we need to enable gcc-8 and g++-8
```

File: scripts/teamcity/20.test.sh [hunk 2]
```diff
@@ -1,5 +1,16 @@
+#!/bin/bash
+
 PYTHON="${PYTHON_CMD:-python}"
 
+# Using cento7 docker we need to enable gcc-8 and g++-8
```
Comment (mkurdej) [inline scripts/teamcity/20.test.sh:5]:
```suggestion
# Using centos7 docker we need to enable gcc-8 and g++-8
```

File: scripts/teamcity/20.test.sh [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+case "$(uname)" in
+    Linux*)
+        source /opt/rh/devtoolset-8/enable
```
Comment (mkurdej) [inline scripts/teamcity/20.test.sh:8]: Isn't it enough to do it once only in `10.build.sh`?

File: scripts/teamcity/20.test.sh [hunk 4]
```diff
@@ -29,6 +40,7 @@ esac
 # first remove system then user
 ${PYTHON} -m pip uninstall -r dev-requirements.txt -y
 ${PYTHON} -m pip uninstall -r dev-requirements.txt -y
```
Comment (mkurdej) [inline scripts/teamcity/20.test.sh:42]: But... that's the same command twice. Why?

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,7 +1,46 @@
...
+FROM centos:7
 
-RUN apt-get update \
-        && apt-get install -y cmake \
-        && apt-get clean
+ARG PYTHON_VERSION
+
+RUN yum install -y \
+
+        # For recent GCC
```
Comment (mkurdej) [inline docker/Dockerfile:9]: I find it error-prone to have blank lines and comments in multi-lines (with trailing `\`).

File: scripts/teamcity/20.test.sh (hunk 3 above)
Comment (vianneyPL) [reply scripts/teamcity/20.test.sh:8]: It is not. It only replaces the current shell.

I could try to put it in the bashrc or something like that.

File: scripts/teamcity/20.test.sh (hunk 4 above)
Comment (vianneyPL) [reply scripts/teamcity/20.test.sh:42]: First one remove from the local user. Second one remove system-wide.

Comment (mkurdej) [reply scripts/teamcity/20.test.sh:42]: But... no need to `sudo` or whatever to remove system-wide?

**Pull request closed without merging**